Cython/Debugger/Cygdb.py

```diff
@@ -48,13 +48,19 @@ def make_command_file(path_to_debug_info, prefix_code='', no_import=False):
             try:
                 # Activate virtualenv, if we were launched from one
                 import os
+                import sys
+                import site
                 virtualenv = os.getenv('VIRTUAL_ENV')
                 if virtualenv:
-                    path_to_activate_this_py = os.path.join(virtualenv, 'bin', 'activate_this.py')
-                    print("gdb command file: Activating virtualenv: %s; path_to_activate_this_py: %s" % (
-                        virtualenv, path_to_activate_this_py))
-                    with open(path_to_activate_this_py) as f:
-                        exec(f.read(), dict(__file__=path_to_activate_this_py))
+                    print("gdb_command_file: Activating virtualenv: %s" % virtualenv)
+                    prev_len = len(sys.path)
+                    if sys.platform == 'win32':
+                        site_packages = os.path.join(virtualenv, 'Lib', 'site-packages')
+                    else:
+                        site_packages = os.path.join(virtualenv, 'lib', 'python%s' % sys.version[:3], 'site-packages')
+                    site.addsitedir(site_packages)
+                    sys.path[:] = sys.path[prev_len:] + sys.path[0:prev_len]
```

Comment:
This is making a bit of an assumption that `site.addsitedir()` appends stuff to the end. That's probably ok, but since it's not documented that way, I think it would still be safer so explicitly insert only the newly added paths at the beginning of (a copy of) the original path. Something like this:
```suggestion
                    orig_path = sys.path[:]
                    site.addsitedir(site_packages)
                    # prepend newly added virtualenv paths to sys.path
                    sys.path[:0] = [path for path in sys.path if path not in orig_path]
```

Reply:
You are right, since appending to `sys.path` is not documented so behaviour of `site.addsitedir()` may change in future. 
I will rebase and force push with your suggestions in a minute.